tests/test_app.py

```diff
@@ -142,17 +142,29 @@ def w():
 
 
 def run_without_tty(
-    args: list[str], env: typing.Optional[dict[str, str]] = None, killtime: int = 15, termtime: int = 10
+    args: list[str],
+    env: typing.Optional[dict[str, str]] = None,
+    killtime: int = 15,
+    termtime: int = 10,
+    *,
+    treat_warnings_as_errors: bool = False,
 ) -> tuple[str, str, int]:
     """Run the given command without a tty"""
+    if treat_warnings_as_errors:
+        if env is None:
+            env = {}
```

Comment:
This is not correct. 

* `env=None` means: run the process with the same environment variables as the parent process.
* `env={}` means: run the process with no environment variables at all. 

So by setting `treat_warnings_as_errors=True` it suddenly runs only with the environment variable `PYTHONWARNINGS` set, but all other environment variables are no longer there.

Reply:
You are right. I overlooked that one.

Reply:
Nice catch, I'll update the message to mention we only log the extra ones, so as to not clutter the output.